Переписываем приложение на Jetpack Compose

il76 opened the pull request:
Сделал по минимальному варианту (1) из-за недостатка времени.
Изначально в https://github.com/il76/Playlist-Maker/compare/dev-27?expand=1 начал делать всё сразу, успел сделать навигацию и большинство экранов, но уже не успевал отладить экран плеера с BottomSheet и привязкой сервиса, а также экран просмотра плейлиста. Поэтому пришлось откатиться и в новой ветке (этой) портировать наработки из той ветки и сделать MVP.

Commit:
Портируем наработки из ветки dev27, которую не успеваю сделать до дедлайна.
Страница настроек на Jetpack. Цвета, шрифты.

Commit:
Страница поиска.

Commit:
Страница медиатеки

Commit:
Вёрстка списков

Commit:
Перенёс Composable функции в отдельные файлы из фрагментов

sukhoikms27 commented on app/src/main/java/com/il76/playlistmaker/media/ui/MediaFragment.kt:
```diff
@@ -4,51 +4,21 @@ import android.os.Bundle
-            when(position) {
-                0 -> tab.text = requireContext().getString(R.string.media_tab_faforite_tracks)
-                1 -> tab.text = requireContext().getString(R.string.media_tab_playlists)
+        return ComposeView(requireContext()).apply {
```
compoВо фрагментах дополнительно лучше менять `ViewCompositionStrategy`: `setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)` - чтобы композиция уничтожалась вместе с вью, а не при каждом отсоединении вью от окна (поведение по умолчанию). Это поможет разгрузить UI, избавив его от лишнего затирания и построения композиций, но при этом не нарушая жизненного цикла вью

sukhoikms27 commented on app/src/main/java/com/il76/playlistmaker/media/ui/PlaylistsScreen.kt:
```diff
@@ -0,0 +1,181 @@
+            .clickable(onClick = {
+                val args = bundleOf("playlistid" to playlist.id)
+                navController.navigate(R.id.fragment_playlist, args)
+                //navController.navigate("playlist/${playlist.id}")
```
Закоментированный код лучше сразу удалять. Хранить закоментированный код в проекте - плохая практика, в случае чего этот код можно восстановить с помощью истории гита

Commit:
Потерял круговой индикатор и плейсхолдеры при загрузке изображений.

Commit:
Потерял скругления у плейлистов

Commit:
Убрал лишнюю локальную переменную для хранения текущего поискового запроса, теперь она во вьюмодели лежит.
Исправлено поведение, при котором при клике по элементе из поиска поиск подменялся на историю.

Commit:
Поправил запрос на список плейлистов - из-за конфликта имён колонок неверно выводились данные.
Не выводилась заглушка об отсутствии данных.

Commit:
Поправил запрос на список плейлистов - из-за конфликта имён колонок неверно выводились данные.
Не выводилась заглушка об отсутствии данных.
Поправил цвета кнопки "Обновить"

il76 commented:
⚠️ Критические замечания:
     1 - на экранах медиатеки пропали заглушки (экран избранного и экран плейлистов)
     2 - пропала возможность переключаться между экранами избранного и плейлистов 
     3 - баг: открываем поиск -> вводим любой запрос с выдачей -> открываем трек -> возвращаемся на экран поиска -> должно отрисовыватьс ясостояние выдачи поиска, а отображается "история поиска" (что не консистентно т.к. в поисковой строке остался текущий поиск)
     
1 - исправлено
3 - исправлено

2 - не могу воспроизвести либо исправлено при других правках. Просьба описать более подробно проблему. Если речь про табы "Избранные треки" и "Плейлисты", то переключение происходит корректно. Но не отображались плейсхолдеры об отсутствии данных - возможно речь была об этом?

sukhoikms27 commented:
> 2 - не могу воспроизвести либо исправлено при других правках. Просьба описать более подробно проблему. Если речь про табы "Избранные треки" и "Плейлисты", то переключение происходит корректно. Но не отображались плейсхолдеры об отсутствии данных - возможно речь была об этом?

речь про свайп между ними

Commit:
Добавил пейджер для эффекта свайпа
